dbt/include/dremio/macros/adapters/apply_grants.sql

```
@@ -16,6 +16,23 @@ limitations under the License.*/
     {{ return(False) }}
 {%- endmacro -%}
 
+{%- macro dremio__split_grantee(grantee) -%}
```

After reading @simonpannek comment, I started thinking about adding validations to all the macros parameters we have. 
Of course, it's not Java and nullability here is not the case, but when we should parse original string, we should have some valid format of it and when the format is incorrect, we should throw an exception.

---

https://github.com/dremio/dbt-dremio/blob/53b4dd370f3cd4cd32bbae82d3b1ade38163f47a/dbt/include/dremio/macros/adapters/apply_grants.sql#L28-L29

We have a verification here already in case the user has a prefix that differs from `user` or `role`. If no prefix is provided, we are defaulting back to `user` for backwards compatibility. Following Simon's suggestion, I plan to add a deprecation warning and, in a future version, expand the verification to be more strict for the prefixes. Do you agree?